src/main/java/com/example/db_task/DBConnection/GetConnection.java

```
@@ -0,0 +1,43 @@
+package com.example.db_task.DBConnection;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.sql.*;
+import java.util.InvalidPropertiesFormatException;
+import java.util.Properties;
+
+public class GetConnection {
+
+    /*Initialize class error logger*/
+    Logger logger = LoggerFactory.getLogger(GetConnection.class);
+
+
+    Connection connection = null;
+
+    public Connection getConnection() {
+
+        Properties prop = new Properties();
+        try {
+            prop.loadFromXML(new FileInputStream("src/main/resources/config/Login.xml"));
+            String url = prop.getProperty("URL");
+            String username = prop.getProperty("username");
+            String password = prop.getProperty("password");
+            connection = DriverManager.getConnection(url, username, password);
+
+        return connection;
+    } catch (SQLException throwable) {
+            logger.error("SQL operation error during connection creation, loading stacktrace:"+ throwable.getMessage());
+        } catch (InvalidPropertiesFormatException e) {
+            logger.error("Invalid XML format error, loading stacktrace:"+ e.getMessage());
+        } catch (FileNotFoundException e) {
+            logger.error("File not found error, loading stacktrace:"+ e.getMessage());
+        } catch (IOException e) {
+            logger.error("I/O operation error, loading stacktrace:"+ e.getMessage());
+        }
+        return null;
```

Is logging the exception and returning `null` is the best way to handle issues here?
These infrastructure configuration issues (which can happen here for example) are probably a blocker for your application, so I would rather wrap them in a runtime one or just directly propagate them back to the caller.

---

Remade into Singleton class and now instead of returning null it throws an exception if connection is null.